Objects/dictobject.c

```diff
@@ -1686,50 +1686,12 @@ PyDict_GetItem(PyObject *op, PyObject *key)
 }
 
 Py_ssize_t
-_PyDict_GetItemHint(PyDictObject *mp, PyObject *key,
-                    Py_ssize_t hint, PyObject **value)
+_PyDict_GetItemSpecialize(PyDictObject *mp, PyObject *key, PyObject **value)
```

Comment:
This doesn't get the item anymore. Nor is it specific to specialization, although that is the only place it is used.
How about `PyDict_LookupIndex`?